src/ppom-controller.ts

```diff
@@ -269,10 +286,18 @@ export class PPOMController extends BaseControllerV2<
   }: {
     chainId: string;
     messenger: PPOMControllerMessenger;
-    provider: any;
+    provider: Provider;
     storageBackend: StorageBackend;
     securityAlertsEnabled: boolean;
-    onPreferencesChange: (callback: (perferenceState: any) => void) => void;
+    onPreferencesChange: (
+      callback: (
+        // TOOD: Replace with `PreferencesState` from `@metamask/preferences-controller`
+        preferencesState: { securityAlertsEnabled: boolean } & Record<
+          string,
+          Json
+        >,
```

Comment:
Wrote temporary `PreferencesState` surrogate to avoid introducing `@metamask/preferences-controller` as a dependency. Added TODO so this would be eventually replaced with the full type.